src/server/requestHandlers.js

```diff
@@ -222,45 +193,85 @@ sjrk.storyTelling.server.saveStoryToDatabase = function (dataSource, binaryRenam
-    // this is used primarily by tests, but may be of use
-    // to client-side components too
-    var binaryRenameMap = {};
-
-    fluid.each(blocks, function (block) {
-        if (block.blockType === "image" || block.blockType === "audio" || block.blockType === "video") {
-            if (block.fileDetails) {
-                // Look for the uploaded file matching this block
-                var mediaFile = fluid.find_if(files, function (file) {
-                    return file.originalname === block.fileDetails.name;
-                });
-
-                // If we find a match, update the media URL. If not, clear it.
-                if (mediaFile) {
-                    sjrk.storyTelling.server.setMediaBlockUrl(block, mediaFile.filename);
-                    binaryRenameMap[mediaFile.originalname] = mediaFile.filename;
+sjrk.storyTelling.server.handleSaveStoryFile = function (request, dataSource, authoringEnabled) {
+    if (authoringEnabled) {
+        var id = request.req.params.id;
+
+        // verify that the story exists and isn't published before continuing
+        dataSource.get({directStoryId: id}).then(function (story) {
+            if (story && !story.published && request.req.file) {
+                // if the file is an image, rotate it based on its EXIF data
+                if (request.req.file.mimetype &&
+                    request.req.file.mimetype.indexOf("image") === 0) {
+
+                    sjrk.storyTelling.server.rotateImageFromExif(request.req.file).then(function () {
+                        request.events.onSuccess.fire(request.req.file.destination + "/" + request.req.file.filename);
+                    }, function (error) {
+                        request.events.onError.fire({
+                            errorCode: error.errorCode,
+                            isError: true,
+                            message: error.message || "Unknown error in image rotation."
+                        });
+                    });
                 } else {
-                    sjrk.storyTelling.server.setMediaBlockUrl(block, null);
+                    request.events.onSuccess.fire(request.req.file.destination + "/" + request.req.file.filename);
                 }
             } else {
-                sjrk.storyTelling.server.setMediaBlockUrl(block, null);
+                request.events.onError.fire({
+                    isError: true,
+                    message: "Error saving file: file was not provided or story is already published"
+                });
             }
-        }
-    });
-
-    return binaryRenameMap;
+        }, function (err) {
+            request.events.onError.fire({
+                isError: true,
+                message: "Error retrieving story with ID " + id,
+                error: err
+            });
+        });
+    } else {
+        request.events.onError.fire({
+            isError: true,
+            message: "Saving is currently disabled."
+        });
+    }

```

Comment:
It might be easier to read this if you move it to the top and just have a simple if condition that returns. 

```javascript
if (!authoringEnabled) { 
    request.events.onError.fire({
        isError: true,
        message: "Saving is currently disabled."
    });
    return;  
}
```